### PR TITLE
backend-client: Fix image retrival and edit

### LINKED ISSUE #624: backend-client: Image retrival failed
**Describe the bug**
The API to fetch an image fails. 

**To Reproduce**
1. Upload image
2. Try to edit image
3. Reaches blank page

**Expected behavior**
Image loads

**Additional context**
The decorater in imagescontroller requires project_id for verification which is not satisfied.

### PR DESCRIPTION
**Pull request by AkMo3**
# Description

This PR add necessary parameters to the request, code changes to accept new request, necessary documentation change, some indentation fixes as well.

This PR fixes the issue of not able to edit images as well.

Fixes #624 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] `npm test` 
- [x] `flask test`

**Test Configuration**:

- OS version: Ubuntu 20.04
- Python 3.8

Also, include screenshots of app for the verification and reviewing purpose.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: labellab-client/src/actions/image.js [hunk 1]
```diff
@@ -40,10 +40,14 @@ export const submitImage = (data, projectId, callback) => {
   }
 }
 
-export const fetchProjectImage = (imageId, callback) => {
+export const fetchProjectImage = (projectId, imageId, callback) => {
   return dispatch => {
+    if (imageId === undefined) {
```
Comment (AkMo3) [inline labellab-client/src/actions/image.js:45]: This seems to be caused when the image is not saved in the database (crop phase) so will not be able to fetch the file from the database without image_id.

**Review by amlannandy — CHANGES_REQUESTED** [review]
Please add a recording showcasing the changes. Basically showing that the functionality is working properly.

File: labellab-client/src/actions/image.js (hunk 1 above)
Comment (AkMo3) [reply labellab-client/src/actions/image.js]: My idea was to only check if it is not initialized, as `!imageId` would be true for `0`. The option best would be to replace with `Number.isInteger()` to be more robust. Will do the change.

Comment (amlannandy) [reply labellab-client/src/actions/image.js]: Id cannot be zero as MySQL starts naming ids for 1. Regardless, your update solution works so it's fine.

Comment (AkMo3) [reply labellab-client/src/actions/image.js]: Yeah, that is true as well :) Are any more changes required?

**Comment by AkMo3** [comment]
@amlannandy Please find the link to demo of working https://drive.google.com/file/d/1y3HuQmHTFpCRirGh26EVUd1qg0dnxCE6/view?usp=sharing

File: labellab-client/src/actions/image.js [hunk 2]
```diff
@@ -40,21 +40,25 @@ export const submitImage = (data, projectId, callback) => {
...
-      .catch(err => {
-        if (err.response) {
-          err.response.data
-            ? dispatch(failure(err.response.data.msg))
-            : dispatch(failure(err.response.statusText, null))
-        }
-      })
+    if (isNaN(imageId) || !Number.isInteger(parseFloat(imageId))) {
+      dispatch(failure('ImageId is not a number'))
+    } else {
```
Comment (amlannandy) [inline labellab-client/src/actions/image.js:47]: Don't use if-else. Adds unnecessary nesting.
Write only an if statement. Leave the rest as it is.

```
if (isNaN(imageId) || !Number.isInteger(parseFloat(imageId))) {
  dispatch(failure('ImageId is not a number'))
  return;
}
```

File: labellab-client/src/actions/image.js [hunk 3]
```diff
@@ -40,21 +40,25 @@ export const submitImage = (data, projectId, callback) => {
...
+      dispatch(request())
```
Comment (amlannandy) [inline labellab-client/src/actions/image.js:48]: Move request dispatch outside the if-else block.
`dispatch(request())` will always trigger. 
Then depending on the rest of the request, either `dispatch(success())` or `dispatch(failure())`

**Review by amlannandy — CHANGES_REQUESTED** [review]
Need a couple more small changes. Rest is fine

File: labellab-client/src/actions/image.js (hunk 2 above)
Comment (AkMo3) [reply labellab-client/src/actions/image.js]: Done ✅

File: labellab-client/src/actions/image.js (hunk 3 above)
Comment (AkMo3) [reply labellab-client/src/actions/image.js]: Done ✅

**Review by amlannandy — APPROVED** [review]

**Comment by amlannandy** [comment]
Nice work 👍🏻

**Pull request merged**